homeassistant/components/switchbee/climate.py

```diff
@@ -0,0 +1,189 @@
+"""Support for SwitchBee climate."""
+from __future__ import annotations
+
+from typing import Any
+
+from switchbee.api import SwitchBeeDeviceOfflineError, SwitchBeeError
+from switchbee.const import (
+    ApiAttribute,
+    ThermostatFanSpeed,
+    ThermostatMode,
+    ThermostatTemperatureUnit,
+)
+from switchbee.device import ApiStateCommand, SwitchBeeThermostat
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SwitchBeeCoordinator
+from .entity import SwitchBeeDeviceEntity
+
+FAN_SB_TO_HASS = {
+    ThermostatFanSpeed.AUTO: FAN_AUTO,
+    ThermostatFanSpeed.LOW: FAN_LOW,
+    ThermostatFanSpeed.MEDIUM: FAN_MEDIUM,
+    ThermostatFanSpeed.HIGH: FAN_HIGH,
+}
+
+FAN_HASS_TO_SB = {
+    FAN_AUTO: ThermostatFanSpeed.AUTO,
+    FAN_LOW: ThermostatFanSpeed.LOW,
+    FAN_MEDIUM: ThermostatFanSpeed.MEDIUM,
+    FAN_HIGH: ThermostatFanSpeed.HIGH,
+}
+
+HVAC_MODE_SB_TO_HASS = {
+    ThermostatMode.COOL: HVACMode.COOL,
+    ThermostatMode.HEAT: HVACMode.HEAT,
+    ThermostatMode.FAN: HVACMode.FAN_ONLY,
+}
+
+HVAC_MODE_HASS_TO_SB = {
```

Comment:
If you add a type hint here, then line 157 can be converted to `assert self.hvac_mode is not None`
```suggestion
HVAC_MODE_HASS_TO_SB: dict[HVACMode | str | None, str] = {
```

Reply:
The same can be done on FAN_HASS_TO_SB:
```python
FAN_HASS_TO_SB: dict[str | None, str]
```